File: src/exercises/simple_linked_list/python/template.py
class Element:
    def __init__(self, value):
        self.value = value
        self.next = None

class SimpleLinkedList:
    def __init__(self, values=None):
        self._head = None
        self._length = 0
        if values:
            for v in values:
                self.push(v)

    def __len__(self):
        return self._length

    def push(self, value):
        # Your code here
        pass

    def pop(self):
        # Your code here
        raise IndexError("list is empty")

    def reversed(self):
        # Your code here
        return self

def list_ops(initial_values: list[int], operations: list[dict]) -> dict:
    # Your code here
    return {}

File: src/exercises/simple_linked_list/go/template.go
package main

type Element struct {
	Value int
	Next  *Element
}

type List struct {
	head *Element
	size int
}

func (l *List) Size() int {
	return l.size
}

func (l *List) Push(element int) {
	// Your code here
}

func (l *List) Pop() (int, error) {
	// Your code here
	return 0, nil
}

func (l *List) Reverse() *List {
	// Your code here
	return l
}

func ListOps(initialValues []int, operations []map[string]interface{}) map[string]interface{} {
	// Your code here
	return map[string]interface{}{}
}
